doc/incubating-plugin.md

```diff
@@ -0,0 +1,273 @@
+
+# Using the incubating plugin
+
+The incubating plugin is in the `apollo-gradle-plugin-incubating`artifact. It is a rewrite of the plugin in Kotlin to make it more maintainable and have better support for multiple services.
+
+```
+buildscript {
+    // Replace
+    // classpath("com.apollographql.apollo:apollo-gradle-plugin")
+
+    // By
+    classpath("com.apollographql.apollo:apollo-gradle-plugin-incubating")
+}
+```
+
+# Differences with the current plugin
+
+### Plugin id is now `com.apollographql.apollo`
+
+We renamed the plugin id from `com.apollographql.android` to `com.apollographql.apollo` to make it clear that the plugin works also for non Android projects. `com.apollographql.android` will be removed in a future revision.
+
+```groovy
+// Instead of
+// apply plugin: 'com.apollographql.android'
+
+// Do this
+apply plugin: 'com.apollographql.apollo'
+```
+
+### Using multiple services
+
+The plugin now requires that you specify multiple services explicitely. If you previously had the following layout:
+
+```bash
+src/main/graphql/com/github/schema.json
+src/main/graphql/com/github/GetRepositories.graphql
+src/main/graphql/com/starwars/schema.json
+src/main/graphql/com/starwars/GetHeroes.graphql
+```
+
+You will need to define 2 services:
+
+```kotlin
+apollo {
+  service("github") {
+    sourceFolder.set("com.github")
```

Comment:
Just double checking, is the _folder_ supposed to be `com.github`? Not `com/github`?

Reply:
Nice catch, thanks!